cases/forms/case.py

```diff
@@ -138,10 +145,44 @@ def clean_type(self):
         type = ActionType.objects.get(id=type)
         return type
 
+    def clean(self):
+        if self._is_time_in_future():
+            raise ValidationError("The time of the action can’t be in the future.")
+        return self.cleaned_data
+
     def save(self, case):
+        self.instance.time = self._get_time_from_cleaned_data()
         self.instance.case = case
         super().save()
 
+    @classmethod
+    def _update_initial_arguments(_, kwargs):
+        updated_initial = {}
+        current_time = localtime(now())
+        updated_initial["date"] = current_time.date
+        updated_initial["action_time"] = current_time.strftime(
+            "%-I:%M %p"
+        )  # E.g. "9:03 AM"
+        kwargs.update(initial=updated_initial)
```

Comment:
I wonder if perhaps it should leave it blank for current time, and only ask to be filled in for time in the past. That way the server can know whether they've provided a time or not (they might have taken a while to fill in the form, I mean), and behave differently as I think we want it to do - e.g. not updating if time given.
Also wonder if if a time is given and action is Case closed/reopened, we error - this means we don't have to worry about closure history, and I don't think it's something they'd ever want to backdate.